### ISSUE TITLE
 fine-tuning the asr transformer transformer lm librispeech pre trained model

### ISSUE THREAD

**Issue by 1311004771** [speechbrain/speechbrain#1957]
### 🚀 The feature

When I was fine-tuning the asr transformer transformer lm librispeech pre trained model, there was an issue, and I received a relatively high ACC but a very abnormal WER

%WER 519.85 [ 681 / 131, 552 ins, 0 del, 129 sub ]
%SER 100.00 [ 9 / 9 ]
Scored 9 sentences, 0 not present in hyp.
================================================================================
ALIGNMENTS

Format:
<utterance-id>, WER DETAILS
<eps> ; reference  ; on ; the ; first ;  line
  I   ;     S      ; =  ;  =  ;   S   ;   D  
 and  ; hypothesis ; on ; the ; third ; <eps>
================================================================================
p232_001, %WER 1000.00 [ 30 / 3, 27 ins, 0 del, 3 sub ]
PLEASE ;  CALL ; STELLA ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ;  <eps>  ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ;  <eps>  ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ;  <eps>  ; <eps> ; <eps> ; <eps> ; <eps> ; <eps>
  S    ;   S   ;   S    ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;    I    ;   I   ;   I   ;   I   ;   I   ;   I   ;    I    ;   I   ;   I   ;   I   ;   I   ;   I   ;    I    ;   I   ;   I   ;   I   ;   I   ;   I  
  I    ; DON'T ;  WANT  ;  YOU  ;   TO  ; THINK ;  THAT ;   I   ; DON'T ;  CARE ;  FOR  ;  YOU  ; BECAUSE ;   I   ; DON'T ;  CARE ;  FOR  ;  YOU  ; BECAUSE ;   I   ; DON'T ;  CARE ;  FOR  ;  YOU  ; BECAUSE ;   I   ; DON'T ;  CARE ;  FOR  ;  YOU 
================================================================================
p232_002, %WER 418.18 [ 46 / 11, 36 ins, 0 del, 10 sub ]
ASK ;  HER  ; <eps> ; <eps> ; TO ; BRING ; THESE ; THINGS ;  WITH ; HER  ; FROM ; THE ; STORE ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ; <eps> ;  <eps>  
 S  ;   S   ;   I   ;   I   ; =  ;   S   ;   S   ;   S    ;   S   ;  S   ;  S   ;  S  ;   S   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;   I   ;    I    
 I  ; DON'T ;  WANT ;  YOU  ; TO ; THINK ;  THAT ;   I    ; DON'T ; MEAN ; WHAT ;  I  ;  SAY  ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ;  WHAT ;   I   ;  SAY  ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ;  THAT ;   I   ; DON'T ;  MEAN ; ANYTHING
================================================================================

The above is part of the testing content, but when I trained the model, the loss decreased normally, and the ACC was also normal.

This is the output log from my testing:
speechbrain.utils.train_logger - Epoch loaded: 4 - test loss: 4.76, test ACC: 9.69e-01, test WER: 5.20e+02

This is the code I modified in the train.py section provided by the recipes. How can I solve this problem?

`if __name__ == "__main__":
    # CLI:
    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
    
    if(torch.cuda.is_available()):
        run_opts['device'] = 'cuda'
    print("device:",run_opts['device'])

    with open(hparams_file) as fin:
        hparams = load_hyperpyyaml(fin, overrides)

    # If --distributed_launch then
    # create ddp_group with the right communication protocol
    sb.utils.distributed.ddp_init_group(run_opts)

    # 1.  # Dataset prep (parsing Librispeech)
    from librispeech_prepare import prepare_librispeech  # noqa

    # Create experiment directory
    sb.create_experiment_directory(
        experiment_directory=hparams["output_folder"],
        hyperparams_to_save=hparams_file,
        overrides=overrides,
    )

    # # multi-gpu (ddp) save data preparation
    # run_on_main(
    #     prepare_librispeech,
    #     kwargs={
    #         "data_folder": hparams["data_folder"],
    #         "tr_splits": hparams["train_splits"],
    #         "dev_splits": hparams["dev_splits"],
    #         "te_splits": hparams["test_splits"],
    #         "save_folder": hparams["output_folder"],
    #         "merge_lst": hparams["train_splits"],
    #         "merge_name": "train.csv",
    #         "skip_prep": hparams["skip_prep"],
    #     },
    # )

    ###############################3
    from speechbrain.pretrained import EncoderDecoderASR
    asr_model = EncoderDecoderASR.from_hparams(source="speechbrain/asr-transformer-transformerlm-librispeech", savedir="/ssdhome/zwb521/speechbrain-develop/recipes/LibriSpeech/ASR/transformer/pretrained_models/asr-transformer-transformerlm-librispeech")
    print(asr_model.mods.keys())
    hparams["tokenizer"] = asr_model.tokenizer
    ################################

    # here we create the datasets objects as well as tokenization and encoding
    (
        train_data,
        valid_data,
        test_datasets,
        tokenizer,
        train_bsampler,
        valid_bsampler,
    ) = dataio_prepare(hparams)

    # We download the pretrained LM from HuggingFace (or elsewhere depending on
    # the path given in the YAML file). The tokenizer is loaded at the same time.
    run_on_main(hparams["pretrainer"].collect_files)
    hparams["pretrainer"].load_collected(device=run_opts["device"])

    # load pretrained module
    ##############################3
    # from speechbrain.pretrained import EncoderDecoderASR
    # asr_model = EncoderDecoderASR.from_hparams(source="speechbrain/asr-transformer-transformerlm-librispeech", savedir="/ssdhome/zwb521/speechbrain-develop/recipes/LibriSpeech/ASR/transformer/pretrained_models/asr-transformer-transformerlm-librispeech")
    # print(asr_model.mods.keys())

    modules1 = {"CNN": asr_model.mods.encoder.cnn, 
           "Transformer": asr_model.hparams.Transformer,
           "compute_features": asr_model.mods.encoder.compute_features, # we use the same features 
           "normalize": asr_model.mods.encoder.normalize,
           "seq_lin": asr_model.hparams.seq_lin,
            "ctc_lin": asr_model.hparams.ctc_lin,
          }
    ################################
    #############################3

    # Trainer initialization
    asr_brain = ASR(
        modules=modules1,
        opt_class=hparams["Adam"],
        hparams=hparams,
        run_opts=run_opts,
        checkpointer=hparams["checkpointer"],
    )

    ##############################
    #require grid
    ##############################

    for module in [asr_brain.modules.CNN, asr_brain.modules.Transformer, asr_brain.modules.seq_lin,asr_brain.modules.ctc_lin]:
        for p in module.parameters():
            p.requires_grad = True
    for module in [asr_brain.modules.normalize,asr_brain.modules.compute_features,asr_brain.modules.CNN, asr_brain.modules.Transformer, asr_brain.modules.seq_lin,asr_brain.modules.ctc_lin]:
        for p in module.parameters():
            p = p.to(run_opts['device'])

    asr_brain.modules.normalize.glob_mean = asr_brain.modules.normalize.glob_mean.to(run_opts['device'])
    asr_brain.modules.normalize.glob_std = asr_brain.modules.normalize.glob_std.to(run_opts['device'])

    ################################3
    ##############################3

    for module in [asr_brain.hparams.valid_search,asr_brain.hparams.test_search]:
        # for p in module.parameters():
        #     p = p.to(run_opts['device'])
        module = module.to(run_opts['device'])

    # # Trainer initialization
    # asr_brain = ASR(
    #     modules=hparams["modules"],
    #     opt_class=hparams["Adam"],
    #     hparams=hparams,
    #     run_opts=run_opts,
    #     checkpointer=hparams["checkpointer"],
    # )

    # adding objects to trainer:
    # asr_brain.tokenizer = hparams["tokenizer"]

    asr_brain.tokenizer = asr_model.tokenizer
    train_dataloader_opts = hparams["train_dataloader_opts"]
    valid_dataloader_opts = hparams["valid_dataloader_opts"]

    if train_bsampler is not None:
        train_dataloader_opts = {
            "batch_sampler": train_bsampler,
            "num_workers": hparams["num_workers"],
        }

    if valid_bsampler is not None:
        valid_dataloader_opts = {"batch_sampler": valid_bsampler}

    # Training
    asr_brain.fit(
        asr_brain.hparams.epoch_counter,
        train_data,
        valid_data,
        train_loader_kwargs=train_dataloader_opts,
        valid_loader_kwargs=valid_dataloader_opts,
    )

    # Testing
    for k in test_datasets.keys():  # keys are test_clean, test_other etc
        asr_brain.hparams.wer_file = os.path.join(
            hparams["output_folder"], "wer_{}.txt".format(k)
        )
        asr_brain.evaluate(
            test_datasets[k],
            max_key="ACC",
            test_loader_kwargs=hparams["test_dataloader_opts"],
        )
`

### Solution outline

please help me
